packages/front-end/pages/integrations/vercel/index.tsx

```diff
@@ -0,0 +1,150 @@
+            postToken();
+          }}
+          autoCloseOnSubmit={false}
+        >
+          <div className="alert alert-warning">
+            <strong>Notice:</strong> A Vercel integration already exists for
+            your organization. By clicking <strong>{`"Continue"`}</strong> you
+            will overwrite the existing integration. Click{" "}
+            <strong>{`"Cancel"`}</strong> to avoid your Vercel integration being
+            overwritten.
+          </div>
+        </Modal>
+      ) : (
+        <Modal open>
+          {envModalOpen && (
+            <EnvironmentModal
+              existing={envModalOpen}
+              close={() => setEnvModalOpen(null)}
+              onSuccess={() => {
+                update;
+                setIntegrationAlreadyExists(false);
+              }}
+            />
+          )}
+          <div>
+            <h4>Generate Environment Variables</h4>
+            <div className="text-muted" style={{ fontSize: "0.8rem" }}>
+              Env vars GROWTHBOOK_KEY and GROWTHBOOK_WEBHOOK_SECRET will be
+              created in GrowthBook and Vercel in the following environments.
+            </div>
+            {gbVercelEnvMap.map((elem, i) => (
+              <div key={`keyMap${i}`} className="d-flex mt-3">
+                <div>
+                  <div>
+                    <strong>Vercel environment:</strong>
+                  </div>
+                  <div>{elem.vercel}</div>
+                </div>
+                <div className="ml-5">
+                  <SelectField
+                    label="GrowthBook environment:"
+                    labelClassName="font-weight-bold"
+                    options={environments.map((env) => ({
+                      label: env.id,
+                      value: env.id,
+                    }))}
+                    initialOption="None"
+                    value={elem.gb}
+                    onChange={(selected) => {
+                      const newMap = [...gbVercelEnvMap];
+                      newMap[i].gb = selected;
+                      setGbVercelEnvMap(newMap);
+                    }}
+                  />
+                </div>
+              </div>
+            ))}
+          </div>
+          <div>
+            <div className="row px-2 justify-content-between">
+              <button
+                onClick={() => setEnvModalOpen({})}
+                className="btn btn-link btn-sm mt-3 col-sm-5"
+              >
+                Create new environment
+              </button>
+              <Button
+                className="mt-3 btn-sm col-sm-5"
+                onClick={async () => {
+                  await apiCall("/vercel/env-vars", {
+                    method: "POST",
+                    body: JSON.stringify({ gbVercelEnvMap }),
+                  });
+                  window.location.href = next as string;
+                }}
+              >
+                Submit
+              </Button>
```

Comment:
Use the built-in `submit` prop for the Modal instead of adding your own button.  Makes it render in the modal footer and automatically shows a loading overlay and catches errors.